pyccl/baryons.py

```diff
@@ -0,0 +1,155 @@
+from . import ccllib as lib
+from .base import CCLAutoreprObject
+from .pyutils import check
+from .pk2d import Pk2D
+from .base import unlock_instance
+import numpy as np
+
+
+class Baryons(CCLAutoreprObject):
+    """`BaryonicEffect` obects are used to include the imprint of baryons
+    on the non-linear matter power spectrum. Their main ingredient is a
+    method `include_baryonic_effects` that takes in a `ccl.Pk2D` and
+    returns another `ccl.Pk2D` object that now accounts for baryonic
+    effects.
+    """
+    name = 'base'
```

Comment:
No need to name the base, it will never be instantiated.

Reply:
To prevent instantiation (emulate abstraction w/o having any abstract methods) you can use the following code:
```python
    def __init__(self):
        if type(self) == A:
            raise TypeError("Base class A cannot be instantiated.")
```

Reply:
Doesn't hurt to name it then.

Reply:
Extra code which is never used.
The reason we rename the other is to be able to do `Baryons.from_name(name)`.

Reply:
Done.